### ISSUE TITLE
Update check running eventhough disabled

### ISSUE THREAD

**Issue by cromefire** [gekware/minecraft-server-hibernation#123]
If came across this error:
```
panic: runtime error: invalid memory address or nil pointer dereference
Dec 05 09:49:21 msh[2595114]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x670308]
Dec 05 09:49:21 msh[2595114]: goroutine 6 [running]:
Dec 05 09:49:21 msh[2595114]: msh/lib/progmgr.checkUpdate({0x6e5b2c, 0x6ea79b}, {0x6e65b6, 0x6}, {0x6e9684, 0x10})
Dec 05 09:49:21 msh[2595114]:         msh/lib/progmgr/progmgr.go:141 +0x4c8
Dec 05 09:49:21 msh[2595114]: msh/lib/progmgr.UpdateManager({0x6e65b6, 0x6})
Dec 05 09:49:21 msh[2595114]:         msh/lib/progmgr/progmgr.go:78 +0x93
Dec 05 09:49:21 msh[2595114]: created by main.main
Dec 05 09:49:21 msh[2595114]:         msh/minecraft-server-hibernation.go:45 +0x152
```

I have the update check disabled so I'd expect the update check disabled ([in the config](https://github.com/gekigek99/minecraft-server-hibernation/blob/5c2e6c2fc71d62ed32de09fa94b9ddd74e58a66e/msh-config.yml#L15)) but it seems to be running and I haven't seen any statements that would hinder it from running. I'm running semi-dev, but since my build there hasn't been any significant change to the `progmgr.go` though. Alternatively a fix for the error would probably also be fine but might proves difficult as I haven't been able to reproduce it so far an is probably due restrictions on internet access.

I might have a look at the error in a few days.

**Comment by gekigek99** [comment]
What version are you using?

`notifyUpdate` specifies only if the update will be notified or not to the user, the update manager function is always running, in the future I'll implement some functionalities to block clients that are using old versions for safety purposes.

It never happened to me, maybe it's related to the peculiar config you are using.

Please keep me updated if you plan on debugging the issue.

**Comment by cromefire** [comment]
> What version are you using?

Basically latest dev

> It never happened to me, maybe it's related to the peculiar config you are using.

It may be blocked from accessing the internet, so there's probably just some unhanded connection error, I'll check the docs, because there's only 2 variable that could cause it.

> in the future I'll implement some functionalities to block clients that are using old versions for safety purposes.

I don't think that would be wise without an override. I myself use it in a fairly sandboxed environment, so there not much that can happen besides the MC Server being deleted or something like that. Also people might want to use it in an offline environment.

**Comment by gekigek99** [comment]
> It may be blocked from accessing the internet, so there's probably just some unhanded connection error, I'll check the docs, because there's only 2 variable that could cause it.

yeah maybe... if you find anything please report it

> I don't think that would be wise without an override. 

What do you mean?

> I myself use it in a fairly sandboxed environment, so there not much that can happen besides the MC Server being deleted or something like that. Also people might want to use it in an offline environment.

I know, but this feature would be good for people who maybe don't use such a safe environment...
and for now it's just an idea, surely I will consider every user case when implementing it.

**Comment by gekigek99** [comment]
on a later analysis this was a bug described in #130, caused by an update server error together with a bug in msh.

the bug has been fixed in 18701ed67970b4b106a32fe3d61bb67402ce84c8
